File: Dante1/Module1.11/Zadanie11.03/main.c
#include <stdio.h>
#include <string.h>
#include <ctype.h>

int marks_counter(const char* text, int* uppercase, int* lowercase, int* whitespace);

int main()
{
    char text[1001];
    printf("Enter: ");

    scanf("%1000[^\n]",text);

    int uppercase = 0, lowercase = 0, whitespace = 0;
    int check = marks_counter(text, &uppercase, &lowercase, &whitespace);
    if (check == 1)
    {
        printf("Incorrect input\n");
        return 1;
    }

    printf("\n%i\n%i\n%i\n",uppercase,lowercase,whitespace);

    return 0;
}

int marks_counter(const char* text, int* uppercase, int* lowercase, int* whitespace)
{
    if (text == NULL || uppercase == NULL || lowercase == NULL || whitespace == NULL)
    {
        return 1;
    }

    int uppercase_counter = 0;
    int lowercase_counter = 0;
    int whitespace_counter = 0;

    for (size_t i = 0; i < strlen(text); i++)
    {
        if (islower((unsigned char)*(text + i)))
        {
            lowercase_counter++;
        }
        else if (*(text+i) == '\t' || *(text+i) == ' ' || *(text+i) == '\n')
        {
            whitespace_counter++;
        }
        else if (isupper((unsigned char)*(text + i)))
        {
            uppercase_counter++;
        }
        /*else if (isspace((unsigned char)*(text + i)))
        {
            whitespace_counter++;
        }*/
        else
        {

        }
    }

    *uppercase = uppercase_counter;
    *lowercase = lowercase_counter;
    *whitespace = whitespace_counter;

    return 0;
}

File: Dante1/Module1.11/Zadanie11.03/main.rs
pub fn marks_counter(s: &str, uppercase: &mut i32,
                     lowercase: &mut i32,
                     whitespace: &mut i32,) -> i32 {

    if s.is_empty() {
        return 1;
    }

    let mut c1: i32 = 0;
    let mut c2: i32 = 0;
    let mut c3: i32 = 0;

    for c in s.chars() {
        if c.is_uppercase() {
            c1 += 1;
        }
        else if c.is_lowercase() {
            c2 += 1;
        }
        else if c.is_whitespace() {
            c3 += 1;
        }
        else {
            // ignore
        }
    }

    *uppercase = c1;
    *lowercase = c2;
    *whitespace = c3;

    0
}

fn main() {
    let mut input: String = String::new();
    std::io::stdin().read_line(&mut input).expect("Incorrect input");

    let mut uppercase: i32 = 0;
    let mut lowercase: i32 = 0;
    let mut whitespaces: i32 = 0;

    if marks_counter(&*input, &mut uppercase, &mut lowercase, &mut whitespaces) != 0 {
        panic!("u r noob");
    }

    println!("{0} {1} {2}", uppercase, lowercase, whitespaces);
}
